src/cascadia/TerminalSettingsEditor/NewTabMenu.cpp

```diff
@@ -0,0 +1,154 @@
+        Automation::AutomationProperties::SetName(AddFolderButton(), RS_(L"NewTabMenu_AddFolderButton/[using:Windows.UI.Xaml.Controls]ToolTipService/ToolTip"));
+        Automation::AutomationProperties::SetName(AddMatchProfilesButton(), RS_(L"NewTabMenu_AddMatchProfilesTextBlock/Text"));
+        Automation::AutomationProperties::SetName(AddRemainingProfilesButton(), RS_(L"NewTabMenu_AddRemainingProfilesButton/[using:Windows.UI.Xaml.Controls]ToolTipService/ToolTip"));
+    }
+
+    void NewTabMenu::OnNavigatedTo(const NavigationEventArgs& e)
+    {
+        _ViewModel = e.Parameter().as<Editor::NewTabMenuViewModel>();
+    }
+
+    void NewTabMenu::FolderPickerDialog_Opened(const IInspectable& /*sender*/, const Controls::ContentDialogOpenedEventArgs& /*e*/)
+    {
+        _ViewModel.CurrentFolderTreeViewSelectedItem(nullptr);
+        _ViewModel.GenerateFolderTree();
+    }
+
+    void NewTabMenu::FolderPickerDialog_PrimaryButtonClick(const IInspectable& /*sender*/, const Controls::ContentDialogButtonClickEventArgs& /*e*/)
+    {
+        // copy selected items first (it updates as we move entries)
+        auto entries = winrt::single_threaded_vector<Editor::NewTabMenuEntryViewModel>();
+        for (const auto& item : NewTabMenuListView().SelectedItems())
+        {
+            entries.Append(item.as<Editor::NewTabMenuEntryViewModel>());
+        }
+
+        // now actually move them
+        _ViewModel.RequestMoveEntriesToFolder(entries, FolderTreeView().SelectedItem().as<Editor::FolderTreeViewEntry>().FolderEntryVM());
+    }
+
+    void NewTabMenu::EditEntry_Clicked(const IInspectable& sender, const RoutedEventArgs& /*e*/)
+    {
+        auto folderVM = sender.as<FrameworkElement>().DataContext().as<Editor::FolderEntryViewModel>();
+        _ViewModel.CurrentFolder(folderVM);
+    }
+
+    void NewTabMenu::ReorderEntry_Clicked(const IInspectable& sender, const RoutedEventArgs& /*e*/)
+    {
+        const auto btn = sender.as<Controls::Button>();
+        const auto entry = btn.DataContext().as<Editor::NewTabMenuEntryViewModel>();
+        const auto direction = unbox_value<hstring>(btn.Tag());
+
+        _ViewModel.RequestReorderEntry(entry, direction == L"Up");
+    }
+
+    void NewTabMenu::DeleteEntry_Clicked(const IInspectable& sender, const RoutedEventArgs& /*e*/)
+    {
+        const auto entry = sender.as<Controls::Button>().DataContext().as<Editor::NewTabMenuEntryViewModel>();
+        _ViewModel.RequestDeleteEntry(entry);
+    }
+
+    safe_void_coroutine NewTabMenu::MoveMultiple_Click(const IInspectable& /*sender*/, const RoutedEventArgs& /*e*/)
+    {
+        co_await FindName(L"FolderPickerDialog").as<Controls::ContentDialog>().ShowAsync();
+    }
+
+    void NewTabMenu::DeleteMultiple_Click(const IInspectable& /*sender*/, const RoutedEventArgs& /*e*/)
+    {
+        // copy selected items first (it updates as we delete entries)
+        auto entries = winrt::single_threaded_vector<Editor::NewTabMenuEntryViewModel>();

```

Comment:
(same here, possibly elsewhere too, but I'll stop commenting so that I don't spam you :D)